src/WebJobs.Extensions.OpenAI/Search/SemanticSearchConverter.cs

```diff
@@ -33,15 +34,25 @@ class SemanticSearchConverter :
     public SemanticSearchConverter(
         OpenAISDK.OpenAIClient openAIClient,
         ILoggerFactory loggerFactory,
-        ISearchProvider searchProvider)
+        IEnumerable<ISearchProvider> searchProviders,
+        IOptions<OpenAIConfigOptions> openAiConfigOptions)
     {
         this.openAIClient = openAIClient ?? throw new ArgumentNullException(nameof(openAIClient));
         this.logger = loggerFactory?.CreateLogger<SemanticSearchConverter>() ?? throw new ArgumentNullException(nameof(loggerFactory));
 
-        // This will be null if no search provider extension is configured
-        // TODO: Eventually we need to resolve this by name at execution time by name so that we can support
-        //       multiple search providers.
-        this.searchProvider = searchProvider;
+        openAiConfigOptions.Value.SearchProvider.TryGetValue("type", out object value);
+        this.logger.LogDebug("Type of the searchProvider configured in host file: {type}", value);
+        
+        if (searchProviders.Count() > 1)
```

Comment:
Generally speaking, we should try to avoid calling `.Count()` on an `IEnumerable<T>`. In this case, I suggest we instead always do `.FirstOrdefault(...)` with a name comparison. If we get back `null`, it should be an error.

Reply:
Just keeping - 
```
if (this.searchProvider == null)
{
    throw new InvalidOperationException(
        "No search provider is configured. Search providers are configured in the host.json file. For .NET apps, the appropriate nuget package must also be added to the app's project file.");
}
```

It is mandatory to specify the type in config now. We are not looking at default or taking first or default in case config is not present